tensorbay/client/segment.py

```diff
@@ -434,20 +434,30 @@ def delete_sensor(self, sensor_name: str) -> None:
 
         self._client.open_api_do("DELETE", "sensors", self.dataset_id, json=delete_data)
 
-    def upload_frame(self, frame: Frame, frame_index: Optional[int] = None) -> None:
-        """Upload frame object to the draft.
+    def upload_frame(self, frame: Frame, upload_time: Optional[ulid.Timestamp] = None) -> None:
```

Comment:
Use `float` rather than `ulid.Timestamp`?

Also the argument name `upload_time` is too explicit. The timestamp of a frame may not be the time of uploading.